apps/docs/src/pages/en/features/completion.md

```diff
@@ -0,0 +1,162 @@
+
+<br>
+
+## Generate Completion Script - ZSH & Bash
+
+Now, you can dynamically generate a completion script for your CLI by running
+the the following command:
+
+```bash
+my-cli completion-script
+```
+
+<hr>
+
+Although, there is no need for that. The best convention will be to load this
+script on shell startup:
+
+**To load the completion script on shell startup**, you can use this command to
+update your `.bashrc` or `.zshrc` file:
+
+```bash
+echo "source <(my-cli completion-script)" >> ~/.bashrc
+```
+
+OR
+
+```bash
+echo "source <(my-cli completion-script)" >> ~/.zshrc
+```
+
+<br>
+
+## Fig Completion integration
+
+After setting up your `CompletionFactory`, you can generate a completion file
+for Fig, and deploy it later. To do so, run the following command:
+
+### Creates a `.fig` directory over your working directory
+
+```bash
+npx @withfig/autocomplete-tools init
+npm --prefix ./.fig/autocomplete run create-spec <CLI_NAME>
+```
+
+### Generate a completion file
+
+```bash
+my-cli generate-fig-spec > ./.fig/autocomplete/src/<CLI_NAME>.ts
+```
+
+If created by running the CLI via an NPM script, check the output file and make
+sure no execution logs exist.
+
+### Build the completion file
+
+```bash
+npm --prefix ./.fig/autocomplete run build
+```
+
+<br>
+<hr>
+<br>
+
+**In order to test it, you need to have the executable CLI file related to the
+completion file you just created, in that way:**
+
+```tree
+.
+├── .fig
+│   └── autocomplete
+│       ├── build
+│       │   └── <CLI_NAME>.js
+│       └── src
+│           └── <CLI_NAME>.ts
+└── <CLI_EXECUTABLE> - same name as <CLI_NAME>
+```
+
+Then, Fig will be available over your CLI by running:
+
+```bash
+./<CLI_NAME> <FIG AVAILABLE HERE!>
+```
+
+<br>
+
+To publish your completion file, execute the following:
+
+```bash
+npm --prefix ./.fig/autocomplete run publish-spec
+```
+
+Now, you'll have Fig completion over your CLI globally! (can be accessed only by
+you - unless you publish it to Fig's community).
+
+You can also generate this completion file as a part of your build process, and
+publish it along with your CLI, so your users will have it by default.
+
+<br>
+
+Read more about Fig completion:
+
+[Private Autocomplete](https://fig.io/docs/guides/private-autocomplete)
+
+[Autocomplete For Internal Tools](https://fig.io/docs/guides/autocomplete-for-internal-tools)
+
+[Create Your CLI Spec](https://fig.io/docs/getting-started) &
+[Upload Your CLI To Fig's Community](https://fig.io/docs/getting-started/contributing)
+
+<br>
+
+## Note
+
+By using the `CompletionFactory`, your CLI might create 3 new commands:
+
+- `completion-script` - Generates the completion script (ZSH & Bash)
+- `completion` - Used by the shell to get the completion, triggered by tab (ZSH
+  & Bash)
+- `generate-fig-spec` - Generates a completion file (Fig)
+
+Please note that these commands do not exist in your CLI, and are only used by
+the `CompletionFactory`.
+
+## Supported Shells
+
+- Bash
+- ZSH
```

Comment:
Does Fig need to be added here?

Reply:
Nope, it's an external app, these are shells

Reply:
Ah, yep, you've got a point there